pkgs/tools/backup/grab-site/default.nix

```diff
@@ -0,0 +1,47 @@
+{ lib
+, python312
+, fetchPypi
+, fetchFromGitHub
+}:
+with python312.pkgs;
+buildPythonApplication rec {
+  pname = "grab-site";
+  version = "2.2.7";
+
+  disabled = pythonOlder "3.12";
+
+  src = fetchFromGitHub {
+    rev = version;
+    owner = "ArchiveTeam";
+    repo = "grab-site";
+    sha256 = "sha256-tf8GyFjya3+TVc2VjlY6ztfjCJgof6tg4an18pz+Ig8=";
+  };
+
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace '"wpull @ https://github.com/ArchiveTeam/ludios_wpull/tarball/master#egg=wpull-${ludios_wpull.version}"' '"wpull"'
+  '';
+
+  propagatedBuildInputs = [
+    click
+    ludios_wpull
+    manhole
+    lmdb
+    autobahn
+    fb-re2
+    websockets
+    faust-cchardet
+  ];
+
+  checkPhase = ''
+    export PATH=$PATH:$out/bin
```

Comment:
isn't  $out is still empty during checkPhase ?

Reply:
That seems to be incorrect, as the `bash ./tests/offline-tests` on the next line fails to execute, complaining about `grab-site` not being in the path. So for now I'm leaving the `PATH` line here.